samagra-comms/uci-web-channel
# Feat: OTP Login and Typescript types added

## Description

- Added the OTP login feature using [APIs](https://github.com/Samarth-HP/esamwad-user-service)
- Added TypeScript Types in all components.

Resolves(#12)

### Changes

- Added two Pages:- `/login` and `/otp`
- Created a small api in `/api/auth` which calls `jwt.verify` to check the validity of the token saved in the local storage

## How to test

1. Start your development server using `yarn run dev` or `npm run dev` and go to the main site and you will be redirected to the `/login` page since there is no access_token (jwt token) in your local storage
![image](https://user-images.githubusercontent.com/76895787/177202165-89dabbe7-013a-43ec-911c-384c0d893820.png)

2. Enter your phone number in the login page and press submit and you will be redirected to the `/otp`
![image](https://user-images.githubusercontent.com/76895787/177202341-f71ea12b-7c81-4e97-9243-40704648be55.png)

3. Enter the OTP you received on your phone number and submit and you will be redirected to the home page and now you can check that you have a access_token in the local storage
![image](https://user-images.githubusercontent.com/76895787/177202458-204b66d8-8480-4ef8-bb80-07966498be1e.png)
![image](https://user-images.githubusercontent.com/76895787/177202479-10fc60d0-bbdf-4cd0-be1a-5f86710287b7.png)
